File: src/main/java/io/dustin/finstage/user/infra/entity/UserEntity.java
package io.dustin.finstage.user.infra.entity;

import jakarta.persistence.*;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Entity
@Table(name = "users")
@Getter
@NoArgsConstructor(access = AccessLevel.PROTECTED)
public class UserEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String email;
    private String password;

    public UserEntity(Long id, String email, String password) {
        this.id = id;
        this.email = email;
        this.password = password;
    }
}

File: src/main/kotlin/io/dustin/finstage/user/infra/entity/UserEntity.kt
package io.dustin.finstage.user.infra.entity

import jakarta.persistence.*

@Entity
@Table(name = "users")
class UserEntity(

        @Id
        @GeneratedValue(strategy = GenerationType.IDENTITY)
        val id: Long? = null,

        val email: String,

        val password: String

) {
    protected constructor() : this(null, "", "")
}
